frontend/src/common/apiManager/httpRequestHandler.ts

```diff
@@ -1,11 +1,24 @@
 import axios from "axios";
+import { v4 as uuidv4 } from "uuid";
 import {
   applyWhenNotNullable,
   getDefaultNullableVal,
   getDefaultRequiredVal,
 } from "../helpers/util";
 import { Nullable, RequiredOrNull } from "../types/common";
 
+// Request interceptor to add a correlationId to the header.
+axios.interceptors.request.use(
+  function (config) {
+    const { headers } = config;
+    headers.set("x-correlation-id", uuidv4());
+    return config;
+  },
+  function (error) {
+    console.log("Unable to make a request:", error);
```

Comment:
do we need this `console.log`? If there's an error, what are we supposed to do about it? eg. maybe in the future it'll be useful for something like HoneyBadger for client-side error capturing

Reply:
It is useful from a debugging standpoint for the devs. It offers no value to an actual user. This is an exceptionally rare situation to even arrive into this block. So, I don't think any effort in this direction is worth it.

I could remove this code block too but that would just mean an undocumented error scenario, however rare it is. Hence why I went with console log.

Reply:
ok